types/src/nibble/nibble_path/mod.rs

```diff
@@ -104,6 +108,32 @@ impl NibblePath {
         NibblePath { num_nibbles, bytes }
     }
 
+    pub fn new_from_state_key(state_key: &StateKey, num_nibbles: usize) -> Self {
+        NibblePath::new_from_byte_array(state_key.hash().to_vec(), num_nibbles)
```

Comment:
nit: no need to `to_vec()` if `new_from_byte_array()` takes `&[u8]` (so we memcopy less)

Reply:
Good point - fixed it.